### PR TITLE
add test files for all tests

### PR DESCRIPTION
**Pull request by Itswali**
# In This PR
* Use jest framework for testing.
* Use the mock storage object you have created in Part 1 of this project to mock the localStorage updates. 
* Mock the HTML elements to test DOM manipulation functions. 
* Make sure you group your tests using the describe() method.

### CODE REVIEW

**Comment by julie-ify** [comment]
Hi @Itswali,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### Kindly ensure your two other coding partners commit in this project since it is collaborative

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/test/edit-todo.test.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import Todo from '../TodoClass.js';
+import createTodoItem from '../CreateTodoItem.js';
+
+describe('Test for edit functionality', () => {
+  test('Edit todo item', () => {
+    const todo = new Todo('Make tests for this!!');
+    todo.updateTodo('Changes made');
+    expect(todo.task).toBe('Changes made');
+    // const todoNode = createTodoItem(todo);
```
Comment (codecaiine) [inline src/test/edit-todo.test.js]: - [ ] Removing commented code not only makes it harder to find later, but it also makes it so people in the future don't know it existed before. For finding the code, there are [tools and git commands](http://stackoverflow.com/q/278192/971592) to help you look at the history of a file. Please remove all commented code and unnecessary files from your code base :smile:

File: src/test/clear-button.test.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    todoCheckbox.click();
+    const todoButton = document.getElementById('clear-button');
+
+    const clearTodos = jest.fn(() => {
+      Todo.clearCompleted();
+      todoContainer.innerHTML = '';
+    });
+
+    todoButton.addEventListener('click', clearTodos);
+    // console.log('completed', todo.isCompleted);
```
Comment (codecaiine) [inline src/test/clear-button.test.js]: - [ ] Removing commented code not only makes it harder to find later, but it also makes it so people in the future don't know it existed before. For finding the code, there are [tools and git commands](http://stackoverflow.com/q/278192/971592) to help you look at the history of a file. Please remove all commented code and unnecessary files from your code base :smile:

File: src/test/clear-button.test.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+// jest.mock('../clearTodos')
+
+import Todo from '../TodoClass.js';
+// import { clearTodos } from '../todoCrud.js';
```
Comment (codecaiine) [inline src/test/clear-button.test.js]: - [ ] Removing commented code not only makes it harder to find later, but it also makes it so people in the future don't know it existed before. For finding the code, there are [tools and git commands](http://stackoverflow.com/q/278192/971592) to help you look at the history of a file. Please remove all commented code and unnecessary files from your code base :smile:

File: src/test/update-status.test.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import Todo from '../TodoClass.js';
+import createTodoItem from '../CreateTodoItem.js';
+
+describe('Make test for status of completed ', () => {
+  test('Update todo items status', () => {
+    const todo = new Todo('Make tests for this!!');
+    todo.changeStatus(true);
+    expect(todo.isCompleted).toBe(true);
+    // const todoNode = createTodoItem(todo);
```
Comment (codecaiine) [inline src/test/update-status.test.js]: - [ ] Removing commented code not only makes it harder to find later, but it also makes it so people in the future don't know it existed before. For finding the code, there are [tools and git commands](http://stackoverflow.com/q/278192/971592) to help you look at the history of a file. Please remove all commented code and unnecessary files from your code base :smile:

File: src/test/clear-button.test.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+// jest.mock('../clearTodos')
```
Comment (codecaiine) [inline src/test/clear-button.test.js]: - [ ] Removing commented code not only makes it harder to find later, but it also makes it so people in the future don't know it existed before. For finding the code, there are [tools and git commands](http://stackoverflow.com/q/278192/971592) to help you look at the history of a file. Please remove all commented code and unnecessary files from your code base :smile:

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @Itswali @admirerbrown @ReemMohamedAbdelfatah , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Good testing implementation :heavy_check_mark: 
- No linters errors :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [ ] [OPTIONAL] In your readme we have a double Table of content section. Please make sure that we have only one section for a table of content :+1: 
- [ ] [OPTIONAL] The link to your license is broken please make sure to fix it and also make sure that all the links in your readme is working :+1: 


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by alexander16108 — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**